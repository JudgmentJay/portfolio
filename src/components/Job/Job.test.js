import React from 'react'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import renderer from 'react-test-renderer'

import { Job } from '../'

beforeEach(() => {
	const mockIntersectionObserver = jest.fn()

	mockIntersectionObserver.mockReturnValue({
		observe: () => null,
		unobserve: () => null,
		disconnect: () => null
	})

	window.IntersectionObserver = mockIntersectionObserver
})

const mockData = {
	title: 'Software Engineer II',
	company: 'Saatva',
	description: 'Full-stack work on a large e-commerce platform. HTML, SASS, JavaScript, React, Node, Express, PHP, GraphQL, SQL.',
	startDate: 'November 2018',
	endDate: 'December 2020',
	img: 'job_saatva',
	testId: 'job-0',
	odd: true
}

it('Should be left-aligned if it\'s an even-numbered job', () => {
	render(<Job {...mockData} odd={false} />)

	const container = screen.getByTestId(mockData.testId)

	expect(container).toHaveClass('jobContainer--left', { exact: false })
})

it('Should be right-aligned if it\'s an odd-numbered job', () => {
	render(<Job {...mockData} />)

	const container = screen.getByTestId(mockData.testId)

	expect(container).toHaveClass('jobContainer--right', { exact: false })
})

it('Renders correctly', () => {
	const tree = renderer
		.create(<Job {...mockData} />)
		.toJSON()

	expect(tree).toMatchSnapshot()
})
