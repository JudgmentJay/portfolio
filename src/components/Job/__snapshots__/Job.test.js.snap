// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  className="jobContainer jobContainer--right"
  data-testid="job-0"
>
  <div
    className="job"
  >
    <div
      className="imgContainer"
    >
      <picture>
        <source
          media="(max-width: 519px)"
          srcSet="img/job_saatva_m.webp"
          type="image/webp"
        />
        <source
          media="(max-width: 519px)"
          srcSet="img/job_saatva_m.jpg"
          type="image/jpeg"
        />
        <source
          media="(max-width: 739px)"
          srcSet="img/job_saatva_t.webp"
          type="image/webp"
        />
        <source
          media="(max-width: 739px)"
          srcSet="img/job_saatva_t.jpg"
          type="image/jpeg"
        />
        <source
          srcSet="img/job_saatva.webp"
          type="image/webp"
        />
        <source
          srcSet="img/job_saatva.jpg"
          type="image/jpeg"
        />
        <img
          alt="Software Engineer II"
          className=""
          loading="lazy"
          sizes="(max-width: 519px) 460px, (max-width: 739px) 100px, 125px"
          src="img/job_saatva.jpg"
          srcSet="img/job_saatva.jpg 460w, img/job_saatva.jpg 100w, 125w"
        />
      </picture>
    </div>
    <div
      className="content"
    >
      <span
        className="date"
      >
        November 2018
        â€“
        December 2020
      </span>
      <h4
        className="title"
      >
        Software Engineer II
      </h4>
      <h5
        className="subtitle"
      >
        Saatva
      </h5>
      <p
        className="description"
      >
        Full-stack work on a large e-commerce platform. HTML, SASS, JavaScript, React, Node, Express, PHP, GraphQL, SQL.
      </p>
    </div>
  </div>
</div>
`;
